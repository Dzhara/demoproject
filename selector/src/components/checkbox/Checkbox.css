.badge {
    background-color: #E5E5E5;
    padding: 3px 0;
    margin: 10px;
    border: 1px solid rgb(163,163,163);
    color: rgb(30,67,110);
    font-weight: 500;
    width: 85px;
    text-align: left;
    display: flex;
    align-items: center;
}


.badge > input {
    margin-left: 5px;
}

.badge:hover,
.badge > input:hover,
.badge > label:hover {
    cursor: pointer;
    background: white;
}

.badge.selected {
    background: white;
}

input:focus {
    outline:none;
}


input[type="checkbox"]{
    -webkit-appearance: initial;
    appearance: initial;
    width: 12px;
    border: 1px solid  rgb(163,163,163);
    border-radius: 2px;
    height: 12px;
    position: relative;
}

input[type="checkbox"]:checked:after {
    content: "X";
    color: red; 
    position: absolute;
    left: 50%;
    top: 50%;
    -webkit-transform: translate(-50%,-50%);
    -moz-transform: translate(-50%,-50%);
    -ms-transform: translate(-50%,-50%);
    transform: translate(-50%,-50%);

}