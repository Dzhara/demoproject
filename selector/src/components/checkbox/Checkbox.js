import React from 'react';
import PropTypes from 'prop-types';
import { Checkbox as CheckboxSemantic } from 'semantic-ui-react';

import './Checkbox.css';
export default class Checkbox extends React.Component {

    constructor(props) {
        super(props);

        this.state = {
            formattedValue: JSON.stringify(this.props.value)
        }
    }


    onChange = (e, { checked }) => {
        this.props.onChange(checked);
    }

    render() {
        const className = ['badge', this.props.checked ? 'selected' : null].join(' ');
        return (
            <CheckboxSemantic
                className={className}
                label={this.props.label}
                value={this.props.formattedValue}
                name={this.props.name}
                checked={this.props.checked}
                onChange={this.onChange}
            />
        );
    }
}

Checkbox.propTypes = {
    onChange: PropTypes.func,
    label: PropTypes.string,
    value: PropTypes.any,
    name: PropTypes.string,
    checked: PropTypes.bool
}

Checkbox.defaultProps = {
    onChange: () => { },
    label: null,
    value: undefined,
    name: null,
    checked: false
}