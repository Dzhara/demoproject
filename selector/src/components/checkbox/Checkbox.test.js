import React from "react";
import renderer from "react-test-renderer";
import Checkbox from "./Checkbox";

describe("Checkbox", () => {
  test("snapshot renders", () => {
    const component = renderer.create(
      <Checkbox label={"test"} value={"test"} checked={true} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
