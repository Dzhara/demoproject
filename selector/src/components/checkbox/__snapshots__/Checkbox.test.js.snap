// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox snapshot renders 1`] = `
<div
  className="ui checked checkbox badge selected"
  onChange={[Function]}
  onClick={[Function]}
  onMouseDown={[Function]}
  onMouseUp={[Function]}
>
  <input
    checked={true}
    className="hidden"
    name={null}
    readOnly={true}
    tabIndex={0}
    type="checkbox"
  />
  <label>
    test
  </label>
</div>
`;
